ravenframework/Models/PostProcessors/SparseSensing.py

```diff
@@ -0,0 +1,190 @@
+    self.pivotParameter = None                               # time-dependent data pivot parameter. None if the problem is steady state
+    self.validDataType = ['PointSet','HistorySet','DataSet'] # FIXME: Should remove the unsupported ones
+    self.sparseSensingGoal = None                            # The goal of the sensor selection. i.e., reconstruction or classification
+    self.nSensors = None                                     # The number of the sensors required by the user.
+    self.nModes = None                                       # The number of modes/basis used to truncate the singular value decomposition
+    self.basis = None                                        # The types of basis used in the projection. i.e., SVD, Identity, or Random Projection
+    self.sensingFeatures = None                              # The variable representing the features of the data i.e., X, Y, SensorID, etc.
+    self.sensingTarget = None                                # The Response of interest to be reconstructed (or classify)
+    self.optimizer = None                                    # The Optimizer type using in the Sparse sensing selection (default: QR)
+    self.sampleTag = 'RAVEN_sample_ID'                       # The sample tag
+
+  def initialize(self, runInfo, inputs, initDict=None):
+    """
+      Method to initialize the DataClassifier post-processor.
+      @ In, runInfo, dict, dictionary of run info (e.g. working dir, etc)
+      @ In, inputs, list, list of inputs
+      @ In, initDict, dict, optional, dictionary with initialization options
+      @ Out, None
+    """
+    super().initialize(runInfo, inputs, initDict)
+    if len(inputs)>1:
+      self.raiseAnError(IOError, 'Post-Processor', self.name, 'accepts only one dataObject')
+
+  def _handleInput(self, paramInput):
+    """
+      Function to handle the parameter input.
+      @ In, paramInput, ParameterInput, the already parsed input.
+      @ Out, None
+    """
+    self.name = paramInput.parameterValues['name']
+    for child in paramInput.subparts:
+      self.sparseSensingGoal = child.parameterValues['subType']
+      self.nSensors = child.findFirst('nSensors').value
+      self.nModes = child.findFirst('nModes').value
+      self.basis = child.findFirst('basis').value
+      self.sensingFeatures = child.findFirst('features').value
+      self.sensingTarget = child.findFirst('target').value
+      self.optimizer = child.findFirst('optimizer').value
+      if child.findFirst('seed') is not None:
+        self.seed = child.findFirst('seed').value
+      else:
+        self.seed = None
+      if child.parameterValues['subType'] not in self.goalsDict.keys():
+        self.raiseAnError(IOError, '{} is not a recognized option, allowed options are {}'.format(child.getName(),self.goalsDict.keys()))
+    _, notFound = paramInput.subparts[0].findNodesAndExtractValues(['nModes','nSensors','features','target'])
+    # notFound must be empty
+    assert(not notFound)
```

Comment:
I think you need to remove the bracket for the assert, otherwise it is always true.

Reply:
I am not sure what you are asking here? Which bracket?

```
>>> assert(not True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
>>> assert(not False)
>>> 
```

Reply:
I changed it to:
```
assert not notFound, "Unexpected nodes in _handleInput"
```